src/shop/tests/test_api.py

```diff
@@ -320,3 +322,47 @@ def test_assign_free_stocktake_chunk(self):
         # Work is finished
         obj5 = api.assign_free_stocktake_chunk(user_obj1.id, stocktake_obj.id)
         self.assertIsNone(obj5)
+
+    def test_predict_quantity(self):
+        initial_qty = 100
+        initial_timestamp = datetime(2016, 11, 14, 0)
+        trx_data = [
+            (-5,  datetime(2016, 11, 15, 0)),
+            (-10, datetime(2016, 11, 16, 0)),
+            (-5,  datetime(2016, 11, 18, 0)),
+            (-5,  datetime(2016, 11, 18, 1)),
+        ]
+        product_obj = factories.ProductFactory.create()
+        # No product transactions yet
+        timestamp = api.predict_quantity(product_obj.id, product_obj.qty - 1)
+        self.assertIsNone(timestamp)
+        factories.ProductTrxFactory.create(
+            product=product_obj,
+            qty=initial_qty,
+            trx_type=enums.TrxType.INVENTORY,
+            date_created=timezone.make_aware(initial_timestamp)
+        )
+        # Product restocked, but no purchases made yet
+        timestamp = api.predict_quantity(product_obj.id, 0)
+        self.assertIsNone(timestamp)
+        for qty, timestamp in trx_data:
+            factories.ProductTrxFactory.create(
+                product=product_obj,
+                qty=qty,
+                date_created=timezone.make_aware(timestamp),
+                trx_type=enums.TrxType.PURCHASE
+            )
+        timestamp = api.predict_quantity(product_obj.id,
+                                         target=product_obj.qty)
+        self.assertEqual(timestamp, date.today())
+        timestamp = api.predict_quantity(product_obj.id, 0)
+        self.assertEqual(timestamp, date(2016, 11, 30))
+
+    @mock.patch('shop.api.predict_quantity')
+    def test_update_quantity_prediction(self, predict_quantity_mock):
+        product_obj = factories.ProductFactory.create()
```

Comment:
Rename this to `product` so we get rid of some of the `_`. Helps with readability.